### PR TITLE
hashlink: init at 1.11

### PR DESCRIPTION
**Pull request by locallycompact**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -5524,6 +5524,12 @@
     githubId = 7622248;
     name = "Sebastian Zivota";
   };
+  locallycompact = {
+    email = "dan.firth@homotopic.tech";
+    github = "locallycompact";
+    githubId = 1267527;
+    name = "Daniel Firth"
```
Comment (dotlambda) [inline maintainers/maintainer-list.nix:5531]:
```suggestion
    name = "Daniel Firth";
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please read https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 116321` at a57bd290 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/2e17e2314f891801a8c5a9bdbaef2669">hashlink</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/compilers/hashlink/default.nix:17:3:
```
   |
17 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (locallycompact) [reply maintainers/maintainer-list.nix]: Ok, I think it's good.

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -5524,6 +5524,12 @@
     githubId = 7622248;
     name = "Sebastian Zivota";
   };
+  locallycompact = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:5527]: Please put this into a separate commit with the message ``maintainers: add locallycompact``.

File: pkgs/development/compilers/hashlink/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  src = fetchFromGitHub {
+    owner = "HaxeFoundation";
+    repo = "hashlink";
+    rev = version;
+    sha256 = "sha256-Mw0AMPK4fdaAdq+BjnFDpo0B9qhTrecD8roLA/JF/a0=";
+  };
+
+  buildInputs =
+    [ libpng libjpeg_turbo libvorbis SDL2 mesa_glu mbedtls openal libuv ];
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/hashlink/default.nix:15]:
```suggestion
  buildInputs = [ libpng libjpeg_turbo libvorbis SDL2 mesa_glu mbedtls openal libuv ];
```

File: pkgs/development/compilers/hashlink/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/lib
+    mkdir -p $out/include
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/hashlink/default.nix:20]:
```suggestion
    mkdir -p $out/bin $out/lib $out/include
```

File: pkgs/development/compilers/hashlink/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    cp hl $out/bin
+    cp libhl.so *.hdll $out/lib
+    cp src/hl.h src/hlc.h src/hlc_main.c $out/include
+  '';
+
+  dontPatchELF = true;
+  dontStrip = true;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/hashlink/default.nix:27]: Every binary should be stripped. Why don't we want to patchelf?

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -10295,6 +10295,8 @@ in
       graphviz = graphviz-nox;
     });
 
+  hashlink = callPackage ../development/compilers/hashlink {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:10298]:
```suggestion
  hashlink = callPackage ../development/compilers/hashlink { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/hashlink/default.nix (hunk 5 above)
Comment (locallycompact) [reply pkgs/development/compilers/hashlink/default.nix:27]: I don't know what this is for. I just copied this from compilers/neko which is the most similar thing I can think of.

**Comment by locallycompact** [comment]
@SuperSandro2000 Since you seem to be knowledgeable, do you happen to know why this set up doesn't seem to be able to find SDL2 at runtime? https://github.com/HeapsIO/heaps/issues/939

File: pkgs/development/compilers/hashlink/default.nix (hunk 5 above)
Comment (locallycompact) [reply pkgs/development/compilers/hashlink/default.nix]: I've removed this

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 Since you seem to be knowledgeable, do you happen to know why this set up doesn't seem to be able to find SDL2 at runtime? [HeapsIO/heaps#939](https://github.com/HeapsIO/heaps/issues/939)

I think it tries to load the libraries dynamically somewhere which does not work on Ubuntu or NixOS by default.

**Comment by jakobrs** [comment]
I tried to package Hashlink a while ago but gave up when I found out that their Makefile and their CMakeLists build *different sets of libraries*, and neither builds all of the packages the other builds. For example, the CMakeLists files don't build `ssl.hdll`, but the Makefile doesn't build `sqlite.hdll`.

These libraries are built by both cmake and make:
- fmt.hdll
- sdl.hdll
- ui.hdll
- uv.hdll

These libraries are built by make, but not cmake:
- mysql.hdll
- openal.hdll
- ssl.hdll
- mesa.hdll (with `MESA=1` in `makeFlags`, although the build seems to be broken)

These libraries are built by cmake, but not make:
- sqlite.hdll

**Comment by tobil4sk** [comment]
Just as an update, `sqlite.hdll` is included in make as of hashlink 1.12, and so `make` is prefered.
https://github.com/HaxeFoundation/hashlink/pull/520
